### PR TITLE
Fix SSO login not working when reCAPTCHA is enabled

### PR DESCRIPTION
**Pull request by hwupathum**
fixes https://github.com/wso2/product-is/issues/14591

### CODE REVIEW

File: apps/authentication-portal/src/main/webapp/basicauth.jsp [hunk 1]
```diff
@@ -63,6 +63,11 @@
         document.getElementById("restartFlowForm").submit();
     }
 
+    function onCompleted() {
+        console.log('captcha completed.');
```
Comment (thivi) [inline apps/authentication-portal/src/main/webapp/basicauth.jsp:67]: Shall we remove this?

File: apps/authentication-portal/src/main/webapp/basicauth.jsp [hunk 2]
```diff
@@ -74,7 +79,19 @@
                     console.warn("Prevented a possible double submit event");
                 } else {
                     e.preventDefault();
-
+                    <%
+                        if (reCaptchaEnabled) {
+                    %>
+                    if (!grecaptcha.getResponse()) {
+                        console.log('captcha not yet completed.');
```
Comment (thivi) [inline apps/authentication-portal/src/main/webapp/basicauth.jsp:86]: Shall we remove this?

**Review by thivi — APPROVED** [review]

File: apps/authentication-portal/src/main/webapp/basicauth.jsp [hunk 3]
```diff
@@ -74,7 +78,17 @@
                     console.warn("Prevented a possible double submit event");
                 } else {
                     e.preventDefault();
-
+                    <%
+                        if (reCaptchaEnabled) {
+                    %>
+                    if (!grecaptcha.getResponse()) {
+                        event.preventDefault(); //prevent form submit
```
Comment (rajithacharith) [inline apps/authentication-portal/src/main/webapp/basicauth.jsp:85]: Format comments.

Comment (hwupathum) [reply apps/authentication-portal/src/main/webapp/basicauth.jsp:85]: Was fixed in https://github.com/wso2/identity-apps/pull/3301

**Pull request merged**